### PR TITLE
dact-2448 compare addresses carefully

### PR DESCRIPTION
**Pull request by AndreKramer7654**
JIRA https://companieshouse.atlassian.net/browse/DACT-2448
Compare addresses carefully as expected fields may not be present in actual data.

### CODE REVIEW

**Review by jditchburn — APPROVED** [review]

File: src/utils/address.ts [hunk 1]
```diff
@@ -15,10 +15,10 @@ export const compareAddress = (addressOne: Address | undefined, addressTwo: Addr
   if (addressOne?.addressLine2?.toLowerCase().trim() !== addressTwo?.addressLine2?.toLowerCase().trim()){
     return false;
   }
-  if (addressOne?.country.toLowerCase().trim() !== addressTwo?.country?.toLowerCase().trim()){
+  if (addressOne?.country?.toLowerCase().trim() !== addressTwo?.country?.toLowerCase().trim()){
```
Comment (okoye-companieshouse) [inline src/utils/address.ts:18]: (addressOne?.country?.toLowerCase().trim() ?? '') !== (addressTwo?.country?.toLowerCase().trim() ?? '')

File: src/utils/address.ts [hunk 2]
```diff
@@ -15,10 +15,10 @@ export const compareAddress = (addressOne: Address | undefined, addressTwo: Addr
...
     return false;
   }
-  if (addressOne?.locality.toLowerCase().trim() !== addressTwo?.locality?.toLowerCase().trim()){
+  if (addressOne?.locality?.toLowerCase().trim() !== addressTwo?.locality?.toLowerCase().trim()){
```
Comment (okoye-companieshouse) [inline src/utils/address.ts:21]: (addressOne?.country?.toLowerCase().trim() ?? '') !== (addressTwo?.country?.toLowerCase().trim() ?? '')

File: src/utils/address.ts (hunk 1 above)
Comment (AndreKramer7654) [reply src/utils/address.ts:18]: The addresses coming from the UI would have to have these fields so flagging a change.

**Review by PhilipGlover — APPROVED** [review]
👍

**Pull request merged**